File: CourseCatalog.Domain/Models/Course.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace CourseCatalog.Domain.Models
{
    public class Course
    {
        public int CourseId { get; set; }

        public string CourseName { get; set; } = null!;

        public int? RoomNumber { get; set; }

        public bool IsDeleted { get; set; }

        public bool Monday { get; set; }

        public bool Tuesday { get; set; }

        public bool Wednesday { get; set; }

        public bool Thursday { get; set; }

        public bool Friday { get; set; }
        public bool Saturday { get; set; }
        public bool Sunday { get; set; }

        public string? ProfessorName { get; set; }

        public string? ProfessorEmail { get; set; }
    }
}

File: CourseCatalog.UI/src/app/models/Course.ts
export class Course {
    public constructor() {
        this.courseId = 0;
        this.courseName = '';
        this.isDeleted = false;
        this.monday = false;
        this.tuesday = false;
        this.wednesday = false;
        this.thursday = false;
        this.friday = false;
        this.saturday = false;
        this.sunday = false;
    }

    courseId: number;
    courseName: string;
    roomNumber?: number;
    isDeleted: boolean;
    monday: boolean;
    tuesday: boolean;
    wednesday: boolean;
    thursday: boolean;
    friday: boolean;
    saturday: boolean;
    sunday: boolean;
    professorName?: string;
    professorEmail?: string;
}
